main.go

```diff
@@ -1691,18 +1691,19 @@ func (tf *transformer) removeUnnecessaryImports(file *ast.File) {
 			return true
 		}
 
-		uses, ok := tf.info.Uses[node].(*types.PkgName)
-		if !ok {
-			return true
+		switch uses := tf.info.Uses[node].(type) {
+		case *types.PkgName:
+			usedImports[uses.Imported().Path()] = true
+		case types.Object:
+			if uses.Pkg() != nil {
```

Comment:
nit: `if pkg := uses.Pkg(); pkg != nil { ... pkg.Path()` to avoid grabbing Pkg again